### PR TITLE
nixos/wireless: add manual network configuration

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change
Requested in #29622 

---

### CODE REVIEW

File: nixos/modules/services/networking/wpa_supplicant.nix [hunk 1]
```diff
@@ -70,6 +73,32 @@ in {
...
+                key_mgmt=WPA-EAP
+                eap=PEAP
+                identity="user@example.com"
+                password="secret"
+              '';
+              description = ''
+                Use this option to configure advanced authentication methods like EAP.
+                See wpa_supplicant.conf(5) for example configurations.
+
+                Mutually exclusive with <varname>psk</varname> and <varname>pskRaw</varname>.
```
Comment (fpletz) [inline nixos/modules/services/networking/wpa_supplicant.nix]: Could you please add an assertion for this?

Comment (rnhmjoj) [reply nixos/modules/services/networking/wpa_supplicant.nix]: @fpletz done.

**Comment by rnhmjoj** [comment]
ping

**Comment by fpletz** [comment]
Thanks!

**Pull request merged**